apps/condo/domains/acquiring/access/AcquiringIntegrationContext.js

```diff
@@ -77,6 +78,29 @@ async function canManageAcquiringIntegrationContexts ({ authentication: { item:
     return await checkAcquiringIntegrationAccessRight(user.id, integrationId)
 }
 
+async function canManageStatusField ({ authentication: { item: user }, existingItem, operation, originalInput }) {
+    // Admin / support can freely change status field
+    if (user.isAdmin || user.isSupport) return true
+
+    // Nobody else can update status field if it's on verification currently
+    if (existingItem && existingItem.status === CONTEXT_VERIFICATION_STATUS) {
+        return false
+    }
+
+    // Service provider org employee should not be able to finish connecting by itself
+    if ('status' in originalInput && originalInput.status === 'Finished') {
+        const orgId = operation === 'create' ? get(originalInput, ['organization', 'connect', 'id']) : existingItem['organization']
+        if (!orgId) {
+            return false
+        }
+        const org = await getById('Organization', orgId)
+
+        return Boolean(org && org.type !== SERVICE_PROVIDER_TYPE)
+    }
+
+    return true
```

Comment:
Is it correct to grant access by default?

Reply:
Field-level access is triggered after model-level access. So we've already filtered all possible actors:
1. admin/support
2. employee
3. integration

All of them had all access to this field by default, since they're passing model-level access.

After that we want to add few restrictions on top for specific scenarios, but keep previous behaviour overall, that's why `true` is default here.

Reply:
Oh, okay. I didn't catch that it is a field access resolver